R/zchunk_L161.nonghg_en_ssp_R_S_T_Y.R

```diff
@@ -0,0 +1,353 @@
+                                                                                  sep = "_")
+
+    # This creates a tibble with the lowest emissions factors for certain grouped categories
+    # This will be used to assign to certain future values for certain SSPs
+    min_EF_policy <- emfact_scaled %>%
+      filter(year == emissions.GAINS_YEARS[1]) %>%
+      na.omit %>%
+      select(-year, -emfact) %>%
+      left_join(emfact_scaled, by = c("GCAM_region_ID", "Non.CO2", "supplysector", "subsector", "stub.technology",
+                                      "GAINS_region", "IIASA_sector", "variable", "region_grouping")) %>%
+      filter(year == emissions.GAINS_YEARS[length(emissions.GAINS_YEARS)]) %>%
+      left_join(coal_so2, by = "GCAM_region_ID") %>%
+      group_by(Non.CO2, supplysector, subsector, stub.technology, variable, year, region_grouping, policy) %>%
+      summarise(emfact = min(emfact, na.rm = TRUE)) %>%
+      ungroup %>%
+      mutate(variable = paste("min", variable, sep = "_")) %>%
+      unite(col = varyearpol, variable, year, policy) %>%
+      spread(varyearpol, emfact)
+
+
+    # Here we assign all future SSP values based on different rules
+    # SSP15 HighMed income: 2030 = 0.75*CLE2030; 2050 = SLE2030; 2100 = 0.75*MFR2030
+    # SSP15 Low income: 2030 = CLE2030; 2050 = W.Eur CLE2030; 2100 = MFR2030
+    # SSP2 HighMed income + strong regulations: 2030 = CLE2030; 2050 = SLE2030; 2100 = min(SLE2030)
+    # SSP2 HighMed income + weak regulations: 2030 = CLE2030; 2050 = min(CLE2030); 2100 = W.Eur SLE2030
+    # SSP2 Low income: 2030 = CLE2020; 2050 = min(CLE2030); 2100 = SLE2030
+    # SSP34 HighMed income: 2030 = CLE2020; 2050 = CLE2030; 2100 = SLE2030
+    # SSP34 Low income: 2030 = CLE2010; 2050 = CLE2030; 2100 = W.Eur CLE2020
+    SSP_EF <- emfact_scaled %>%
+      # This is to have same starting point as old code, which begins with NA omitted 2010 values
+      filter(year == emissions.GAINS_YEARS[1]) %>%
+      na.omit %>%
+      select(-year, -emfact) %>%
+      # Rejoining in all data, with no NAs in 2010 values
+      left_join(emfact_scaled, by = c("GCAM_region_ID", "Non.CO2", "supplysector", "subsector", "stub.technology",
+                                      "GAINS_region", "IIASA_sector", "variable", "region_grouping")) %>%
+      unite(col = varyear, variable, year) %>%
+      spread(varyear, emfact) %>%
+      repeat_add_columns(SSP_groups) %>%
+      repeat_add_columns(tibble(year = c(2010, 2030, 2050, 2100))) %>%
+      # Adding in policies and other emissions factors
+      left_join_error_no_match(coal_so2, by = "GCAM_region_ID") %>%
+      left_join(marker_region_df, by = c("Non.CO2", "supplysector", "subsector", "stub.technology", "IIASA_sector")) %>%
+      left_join(min_EF_policy, by = c("Non.CO2", "supplysector", "subsector", "stub.technology", "region_grouping")) %>%
+      # Initially set all values to CLE_2010, then change all years greater than 2010
+      mutate(value = CLE_2010,
```

Comment:
I'll do it now. It's probably gonna be a bit uglier than Ben's example because of the Western Europe and min rules, but I agree that this is preferable.